src/png.imageio/pnginput.cpp

```diff
@@ -162,7 +162,8 @@ PNGInput::open(const std::string& name, ImageSpec& newspec)
     bool ok = PNG_pvt::read_info(m_png, m_info, m_bit_depth, m_color_type,
                                  m_interlace_type, m_bg, m_spec,
                                  m_keep_unassociated_alpha);
-    if (!ok || m_err) {
+    if (!ok || m_err
+        || check_open(m_spec, { 0, 1 << 30, 0, 1 << 30, 0, 1, 0, 4 })) {
```

Comment:
I believe there is a bug here: you need `!check_open()`, right? That's probably why CI is failing.

Reply:
Oh yes! First time doing this, forgot there is `make test`.